src/zkevm_specs/evm/execution/slt_sgt.py

```diff
@@ -0,0 +1,49 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+
+from zkevm_specs.encoding import U256, u256_to_u8s
+
+def scmp(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    is_sgt, _ = instruction.pair_select(opcode, Opcode.SGT, Opcode.LGT)
+
+    a = instruction.stack_pop()
+    b = instruction.stack_pop()
+    c = instruction.stack_push()
+
+    a8s = u256_to_u8s(U256(a))
+    b8s = u256_to_u8s(U256(b))
+    c8s = u256_to_u8s(U256(c))
+
+    # if both a and b are unsigned
+    if a8s[0] < 128 and b8s[0] < 128:
+        instruction.constrain_equal(
+            instruction.select(is_sgt, b < a, a < b),
```

Comment:
You cannot directly compare `b` and `a` as they are in the Fp field. You need to first use `rlc_to_bytes` to convert them back to u8 sequence and write constraints using u8s.